floor_generator/lib/processor/type_converter_processor.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:floor_generator/misc/type_utils.dart';
+import 'package:floor_generator/processor/processor.dart';
+import 'package:floor_generator/value_object/type_converter.dart';
+import 'package:source_gen/source_gen.dart';
+
+class TypeConverterProcessor extends Processor<TypeConverter> {
+  final ClassElement _classElement;
+  final TypeConverterScope _typeConverterScope;
+
+  TypeConverterProcessor(
+    final ClassElement classElement,
+    final TypeConverterScope typeConverterScope,
+  )   : _classElement = classElement,
+        _typeConverterScope = typeConverterScope;
+
+  @override
+  TypeConverter process() {
+    final typeArguments = _classElement.supertype.typeArguments;
+    final fieldType = typeArguments[0];
+    final databaseType = typeArguments[1];
+
+    if (!databaseType.isDefaultSqlType) {
+      throw InvalidGenerationSourceError(
+        'Type converters have to convert to a database-compatible type.',
+        todo:
+            'Make the class convert to either int, double, String, bool or Uint8List.',
```

Comment:
Update: I think we should keep bool there, just because it is also a special type in sqlparser and it could come in handy in the future. But I'm still interested in your thoughts :)

Reply:
Let's leave our bool handling as it is 👍 
About introducing default type converters: I'm not a big fan of that as it hides functionality which might lead to unexpected behavior. On the other hand, we basically have a type converter built into the library as we're converting bools everywhere 🙃